### ISSUE TITLE
[BUG] Unable to provision RKE2 cluster from rancher, stuck in provisioning on bootstrap node 

### ISSUE THREAD

**Issue by TachunLin** [harvester/harvester#1954]
**Describe the bug**

Unable to provision RKE2 cluster from rancher `v2.6-head`, keep in `Provisioning` status with message 
This issue exists on both RKE2 `1.22.7+rke2r1` and `1.21.10+rke2r1` 

```
Non-ready bootstrap machine(s) rke2-ubuntu-pool1-bcfd5cfbb-2pj9w and join url to be available on bootstrap node
```

![image](https://user-images.githubusercontent.com/29251855/156367737-43717d56-b5be-43ac-be52-724a2cc3cd57.png)

Further checking cluster node status in RKE2 VM, it looks like node have taints to prevent component to finish deployment.

```
# ssh to rke2 virtual machine

export KUBECONFIG=/etc/rancher/rke2/rke2.yaml
/var/lib/rancher/rke2/bin/kubectl get nodes -o yaml
```

```
spec:
    podCIDR: 10.42.0.0/24
    podCIDRs:
    - 10.42.0.0/24
    taints:
    - effect: NoSchedule
      key: node.cloudprovider.kubernetes.io/uninitialized
      value: "true"
```

**To Reproduce**
1. Create a one node or three nodes harvester cluster 
2. Prepare SLES JeOS image
`SLES15-SP3-JeOS.x86_64-15.3-OpenStack-Cloud-GM.qcow2`
4. Enable virtual network to `harvester-mgmt`
5. Create virtual network `vlan1` with id `1`
6. Import harvester in rancher v2.6-head
7. Create cloud credential 
8. Provision a RKE2 cluster with SLES JeOS image  
![image](https://user-images.githubusercontent.com/29251855/156369683-d94ba2be-1ab5-4df8-9328-013dda470583.png)
9. Check the RKE2 cluster provisioning to `Ready`

**Expected behavior**

Can provision RKE2 cluster in harvester correctly in acceptable time for major OS image (e.g SLES, Ubuntu) 

**Support bundle**

[supportbundle_74b92604-c442-4d1d-a70d-7313e32ac47c_2022-03-02T13-14-35Z.zip](https://github.com/harvester/harvester/files/8170323/supportbundle_74b92604-c442-4d1d-a70d-7313e32ac47c_2022-03-02T13-14-35Z.zip)

**Rancher container log**
[rancher2.6-head_rke2_sles.log](https://github.com/harvester/harvester/files/8170358/rancher2.6-head_rke2_sles.log)

**Environment:**
 - Harvester ISO version: `master-e214d829-head` (03/01)
 - Underlying Infrastructure: one or 3 nodes harvester on local kvm machine
 - Rancher version: v2.6-head (`1d2b746`)

#### Related issue
https://github.com/rancher/rancher/issues/36716

**Comment by noahgildersleeve** [comment]
I tested this and it looks like it's an issue with 1.22.7 not 1.21.10. @TachunLin Could you check too when you get a chance?

**Comment by TachunLin** [comment]
Hi @noahgildersleeve , 
I have checked can provision RKE2 cluster on latest rancher `v2.6-head` (`33db2f5`) with `1.21.10+rke2r1`.
Thank you for the follow up reminder

![image](https://user-images.githubusercontent.com/29251855/156926447-0046d0bd-da70-4965-a795-287f91a9b5a0.png)

![image](https://user-images.githubusercontent.com/29251855/156926066-a6dfaf9c-8843-4f3d-a233-8191546ee08d.png)

### Test Environment:

- Harvester ISO version: master-088a020f-head (03/05)
- Underlying Infrastructure: 1 node harvester on local kvm machine
- Rancher version: v2.6-head (`33db2f5`)

**Comment by FrankYang0529** [comment]
I get different results for different OS with different RKE2 versions.

|       | 1.21.10+rke2r1    | 1.22.7+rke2r1                         |
|-------|-------------------|---------------------------------------|
| Focal | Work              | RKE2 starts but has NoSchedule taint. |
| JeOS  | etcd can't start. | RKE2 starts but has NoSchedule taint. |

### JeOS with 1.21.10+rke2r1

I can fix this one by installing `apparmor-parser`. This one may be related to https://github.com/rancher/rke2/issues/1650. After installing `apparmor-parser`, RKE2 starts without the `NoSchedule` taint, and the cluster becomes active.

![Screen Shot 2022-03-07 at 2 39 07 PM](https://user-images.githubusercontent.com/4927361/156980780-de4e864f-8e6a-45e0-8536-43e57673415f.png)

```
zypper ar --priority=200 http://download.opensuse.org/distribution/leap/15.3/repo/oss repo-oss
zypper ref
zypper in apparmor-parser
```

### Focal/JeOS with 1.22.7+rke2r1

I try to remove the `NoSchedule` taint on Node. Although all pods can run, the cluster status is still in `Provisioning`.

**Comment by TachunLin** [comment]
Confirmed I encounter the same result as @FrankYang0529 https://github.com/harvester/harvester/issues/1954#issuecomment-1060241440 (On rancher `v2.6-head`)

* On 1.22.7+rke2r1, we `can't` provision RKE2 cluster with `both` `Ubuntu cloud focal` and `SLES JeOS` images 
* On 1.21.10+rke2r1, we `can` provision RKE2 cluster on `Ubuntu focal` 
* On 1.21.10+rke2r1, we `can't` provision RKE2 cluster on `SLES JeOS`, but with manual install apparmor-parser can workaround it

**Comment by guangbochen** [comment]
We can convert the `SLES JeOS` issue into a separate one, and this will require proper upstream fix from the SUSE JeOS team, thanks.

**Comment by guangbochen** [comment]
~~This issue is currently blocked by https://github.com/rancher/rancher/issues/36716 and requires RKE2 with v1.22.x  to bump its `calico-typha` to tolerate `node.cloudprovider.kubernetes.io/uninitialized` when a cluster has cloud-provider enabled.~~
Should be fixed via https://github.com/harvester/harvester/issues/1999 with the newest KDM release of Rancher v2.6.4.

**Comment by TachunLin** [comment]
Verified **fixed** on Rancher `v2.6-head 2642acc` (03/16) with Harvester `v1.0-591d8015-head` (03/16). Close this issue.



### Result
Can provision RKE2 cluster with kubernetes version **`v1.21.7+rke2r2`** for the following OS image:
1. Ubuntu focal image
2. SLES JeOS image

![image](https://user-images.githubusercontent.com/29251855/158577116-0bda5f89-2083-49e9-8f55-d6d172df05ef.png)
![image](https://user-images.githubusercontent.com/29251855/158577306-996c53dd-a7e1-43b2-af59-97896faf47f8.png)

### Test Information
* Test Environment: 1 nodes harvester on local kvm machine
* Harvester version: v1.0-591d8015-head (03/16)
* Rancher version: v2.6-head 2642acc
* Kubernetes version: v1.21.7+rke2r2
* OS image: Ubuntu focal, SLES JeOS

### Verify Steps
1. Create a one node harvester 
2. Upload Ubuntu focal and SLES JeOS image
3. Create virtual network vlan1 with id 1
4. Import Harvester from Rancher
5. Create cloud credential 
6. Provision an RKE2 cluster 
7. Add the following use cloud-init data 
```
  password: 123456
  chpasswd: { expire: False }
  ssh_pwauth: True
  ```
8. Select Ubuntu focal or SLES JeOS image
9. Provide available cpu and memory resources
10. Provision the RKE2 cluster
